src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -357,7 +402,7 @@ await _sqlRetryService.ExecuteSql(
                                 new HashingSqlQueryParameterManager(new SqlQueryParameterManager(sqlCommand.Parameters)),
                                 _model,
                                 _schemaInformation,
-                                _reuseQueryPlans.IsEnabled(_sqlRetryService),
+                                reuseQueryPlans,
```

Comment:
Why this change?

Reply:
This allows the setting to use or not use query plans to be controlled from an input parameter, which is then controlled by the header value parsing code.